addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -143,6 +143,26 @@ def _timesheet_prepare_line_values(self, index, employee_id, work_hours_data, da
             'company_id': employee_id.company_id.id,
         }
 
+    def _restore_public_holiday_timesheets(self, employee_ids):
+        vals_list = []
+        work_hours_data = self._work_time_per_day()
+        for leave in self:
+            for employee in employee_ids:
+                work_hours_list = work_hours_data[leave.id]
+                for index, (day_date, work_hours_count) in enumerate(work_hours_list):
+                    existing_timesheets = leave["timesheet_ids"].filtered(lambda r: r.employee_id.id == employee.id and r.date == day_date)
+                    if not existing_timesheets:
+                        vals_list.append(
+                            leave._timesheet_prepare_line_values(
+                                index,
+                                employee,
+                                work_hours_list,
+                                day_date,
+                                work_hours_count
+                            )
+                        )
+        return self.env['account.analytic.line'].sudo().create(vals_list)
```

Comment:
```suggestion
    def _generate_public_time_off_timesheets(self, employees):
        timesheet_vals_list = []
        work_hours_data = self._work_time_per_day()
        for leave in self:
            for employee in employees:
                work_hours_list = work_hours_data[leave.id]
                for index, (day_date, work_hours_count) in enumerate(work_hours_list):
                    existing_timesheets = leave.timesheet_ids.filtered(
                        lambda r: r.employee_id.id == employee.id and r.date == day_date)
                    if existing_timesheets:
                        continue
                    timesheet_vals = leave._timesheet_prepare_line_values(
                        index,
                        employee,
                        work_hours_list,
                        day_date,
                        work_hours_count
                    )
                    timesheet_vals_list.append(timesheet_vals)
        return self.env['account.analytic.line'].sudo().create(timesheet_vals_list)
```

Reply:
Thanks for your feedback, I've applied the suggested changes.